### PR TITLE
Disable share view completely when server does not support/has disabled file sharing

### LINKED ISSUE #5826: [Bug]: Share options are displayed when the "File sharing" app is disabled
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

When the "File sharing" app on the Nextcloud server is disabled, you still have the option to share files in the client. If you try to do so, you get a non-speaking error (see attachment below).

### Steps to reproduce

1. Select any file in the Nextcloud folder
2. Right-click -> Nextcloud -> Copy public link
3. An error is displayed

### Expected behavior

The share options in the right-click menu and the client itself shouldn't be visible when "File sharing" is disabled.

### Which files are affected by this bug

"Neues Textdokument.txt" or any other file

### Operating system

Windows

### Which version of the operating system you are running.

Windows 10 Enterprise

### Package

Other

### Nextcloud Server version

26.0.2

### Nextcloud Desktop Client version

3.9.0

### Is this bug present after an update or on a fresh install?

Fresh desktop client install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [X] SSO - SAML
- [ ] Other

### Nextcloud Server logs

```shell
there are no related errors on the server
```

### Additional info

share

[Debug archive](https://github.com/nextcloud/desktop/files/11797652/log.zip)

### PR DESCRIPTION
**Pull request by claucambra**
Fixes #5826

### CODE REVIEW

**Review by mgallien — COMMENTED** [review]
@claucambra see my comment

**Review by mgallien — CHANGES_REQUESTED** [review]
DCO check is failed
@claucambra please have a look

File: src/gui/filedetails/filedetails.cpp [hunk 1]
```diff
@@ -62,6 +62,7 @@ void FileDetails::setLocalPath(const QString &localPath)
     connect(&_fileWatcher, &QFileSystemWatcher::fileChanged, this, &FileDetails::refreshFileDetails);
 
     const auto folder = FolderMan::instance()->folderForPath(_localPath);
+    Q_ASSERT(folder);
```
[the remark this answers is not in the document]
Comment (claucambra) [reply src/gui/filedetails/filedetails.cpp]: Ok, replaced with a if check and return

**Review by mgallien — APPROVED** [review]

**Pull request merged**